ratatui-widgets/src/table.rs

```diff
@@ -1011,6 +1109,23 @@ fn ensure_percentages_less_than_100(widths: &[Constraint]) {
     }
 }
 
+/// Return a `Rect` that represents the area that a `Cell` extends over
+///
+/// Currently, `Cell`s may extend over multiple columns but not multiple rows. This function
+/// returns a `Rect` that spans over a single row and possibly multiple columns, with the `x` field
+/// representing the `Cell`'s start coordinate.
+///
+/// Future extensions that allow a `Cell` to span multiple rows can replace all calls to this
+/// function with an initializer for all four fields of `Rect`.
+const fn create_cell_area(x: u16, width: u16) -> Rect {
+    Rect {
+        x,
+        y: 0,
+        width,
+        height: 0,
+    }
+}
+
```

Comment:
I don't see the value in this right now - it makes the code more difficult to understand for now, so calling YAGNI and suggesting to remove it until we need it. WDYT?

Reply:
Removed this function in 88f1eb37